### PR TITLE
Implementing Delete messages for everyone (group)

### PR DESCRIPTION
**Pull request by PurpShell**
This automatically implements deleting a message for everyone as an admin in the group. Just make sure to provide the fromMe property (fromMe: true in a group means sender_revoke, otherwise if the message isn't sent by you it will do an admin_revoke) or this will break

### CODE REVIEW

**Comment by Kuroosh** [comment]
Thank you!!! ♥

**Comment by PurpShell** [comment]
@adiwajshing ready for review

File: src/Socket/messages-send.ts [hunk 1]
```diff
@@ -620,7 +620,12 @@ export const makeMessagesSocket = (config: SocketConfig) => {
 				const additionalAttributes: BinaryNodeAttributes = { }
 				// required for delete
 				if(isDeleteMsg) {
-					additionalAttributes.edit = '7'
+					// if the chat is a group, and I am not the author, then delete the message as an admin
+					if(content.delete?.remoteJid?.endsWith('@g.us') && !content.delete?.fromMe) {
```
Comment (adiwajshing) [inline src/Socket/messages-send.ts:624]: should use `isJidGroup` here

Comment (adiwajshing) [reply src/Socket/messages-send.ts:624]: Otherwise, looks good to me

Comment (PurpShell) [reply src/Socket/messages-send.ts]: Didn't know about that, thanks

**Comment by PurpShell** [comment]
@adiwajshing resolved

**Comment by AliAryanTech** [comment]
@adiwajshing when pulling updates?

**Comment by Kuroosh** [comment]
> @adiwajshing when pulling updates?

Please Pull :D ♥

**Comment by AliAryanTech** [comment]
> This automatically implements deleting a message for everyone as an admin in the group. Just make sure to provide the fromMe property (fromMe: true in a group means sender_revoke, otherwise if the message isn't sent by you it will do an admin_revoke) or this will break

Bruh, can you give a example code?
How to use that admin delete feature!!

**Comment by PurpShell** [comment]
> > This automatically implements deleting a message for everyone as an admin in the group. Just make sure to provide the fromMe property (fromMe: true in a group means sender_revoke, otherwise if the message isn't sent by you it will do an admin_revoke) or this will break
> 
> Bruh, can you give a example code? How to use that admin delete feature!!

just delete the same way, but if you aren't the one sending the message and it is in a group, an admin delete message is sent instead

**Comment by AliAryanTech** [comment]
> > > This automatically implements deleting a message for everyone as an admin in the group. Just make sure to provide the fromMe property (fromMe: true in a group means sender_revoke, otherwise if the message isn't sent by you it will do an admin_revoke) or this will break
> > 
> > 
> > Bruh, can you give a example code? How to use that admin delete feature!!
> 
> just delete the same way, but if you aren't the one sending the message and it is in a group, an admin delete message is sent instead

Me & Bot admin... 3rd person message
I quoted that message with eval.

```$eval
let ok = M.quoted;
sock.sendMessage(M.from, { delete: ok.key })
```
But, still it didn't delete that message only deleting bot message..
I'm using baileys updated version ^4.3.0

**Comment by PurpShell** [comment]
> > > > This automatically implements deleting a message for everyone as an admin in the group. Just make sure to provide the fromMe property (fromMe: true in a group means sender_revoke, otherwise if the message isn't sent by you it will do an admin_revoke) or this will break
> > > 
> > > 
> > > Bruh, can you give a example code? How to use that admin delete feature!!
> > 
> > 
> > just delete the same way, but if you aren't the one sending the message and it is in a group, an admin delete message is sent instead
> 
> Me & Bot admin... 3rd person message I quoted that message with eval.
> 
> ```
> let ok = M.quoted;
> sock.sendMessage(M.from, { delete: ok.key })
> ```
> 
> But, still it didn't delete that message only deleting bot message.. I'm using baileys updated version ^4.3.0

use github version, hasn't been merged to npm

**Comment by AliAryanTech** [comment]
> > > > > This automatically implements deleting a message for everyone as an admin in the group. Just make sure to provide the fromMe property (fromMe: true in a group means sender_revoke, otherwise if the message isn't sent by you it will do an admin_revoke) or this will break
> > > > 
> > > > 
> > > > Bruh, can you give a example code? How to use that admin delete feature!!
> > > 
> > > 
> > > just delete the same way, but if you aren't the one sending the message and it is in a group, an admin delete message is sent instead
> > 
> > 
> > Me & Bot admin... 3rd person message I quoted that message with eval.
> > ```
> > let ok = M.quoted;
> > sock.sendMessage(M.from, { delete: ok.key })
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But, still it didn't delete that message only deleting bot message.. I'm using baileys updated version ^4.3.0
> 
> use github version, hasn't been merged to npm

Bot Crushed after running repo
run: ```yarn add github:adiwajshing/baileys```

Logs:
[Logs File](https://github.com/adiwajshing/Baileys/files/9310936/logs.txt)

**Comment by Kuroosh** [comment]
> > > > > > This automatically implements deleting a message for everyone as an admin in the group. Just make sure to provide the fromMe property (fromMe: true in a group means sender_revoke, otherwise if the message isn't sent by you it will do an admin_revoke) or this will break
> > > > > 
> > > > > 
> > > > > Bruh, can you give a example code? How to use that admin delete feature!!
> > > > 
> > > > 
> > > > just delete the same way, but if you aren't the one sending the message and it is in a group, an admin delete message is sent instead
> > > 
> > > 
> > > Me & Bot admin... 3rd person message I quoted that message with eval.
> > > ```
> > > let ok = M.quoted;
> > > sock.sendMessage(M.from, { delete: ok.key })
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > But, still it didn't delete that message only deleting bot message.. I'm using baileys updated version ^4.3.0
> > 
> > 
> > use github version, hasn't been merged to npm
> 
> Bot Crushed after running repo run: `yarn add github:adiwajshing/baileys`
> 
> Logs: [Logs File](https://github.com/adiwajshing/Baileys/files/9310936/logs.txt)

what is `M.quoted` ? 
does it contains `.key` ?

**Comment by AliAryanTech** [comment]
> 

**M.quoted** It contains:
```
{
	"sender": {
		"username": "User",
		"jid": "xxxxxxxxx42@s.whatsapp.net"
	},
	"message": {
		"conversation": "\"group\""
	},
	"type": "conversation",
	"hasSupportedMediaMessage": false,
	"content": "\"group\"",
	"key": {
		"remoteJid": "xxxxxxxxxxxx705@g.us",
		"fromMe": true,
		"id": "BAE58CRIF677C6AA",
		"participant": "xxxxxxx42@s.whatsapp.net"
	}
}
```
and **M.quoted.key** contains:
```
{
	"remoteJid": "xxxxxxxxxxxxxx799@g.us",
	"fromMe": true,
	"id": "BAE59A48RUSA9A4A",
	"participant": "xxxxxxx42@s.whatsapp.net"
}
```

**Comment by AliAryanTech** [comment]
> > > > > This automatically implements deleting a message for everyone as an admin in the group. Just make sure to provide the fromMe property (fromMe: true in a group means sender_revoke, otherwise if the message isn't sent by you it will do an admin_revoke) or this will break
> > > > 
> > > > 
> > > > Bruh, can you give a example code? How to use that admin delete feature!!
> > > 
> > > 
> > > just delete the same way, but if you aren't the one sending the message and it is in a group, an admin delete message is sent instead
> > 
> > 
> > Me & Bot admin... 3rd person message I quoted that message with eval.
> > ```
> > let ok = M.quoted;
> > sock.sendMessage(M.from, { delete: ok.key })
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But, still it didn't delete that message only deleting bot message.. I'm using baileys updated version ^4.3.0
> 
> use github version, hasn't been merged to npm

Any solution? 🤔

**Pull request merged**